### DISCUSSION TITLE
Can mockall_double be in dev-dependencies?

### DISCUSSION THREAD

**Discussion by oaijiao** [asomers/mockall#548] [Questions] [answered]
In the mockall_double README, it states mockall_double must be a full dependency. 

This seems to be working even if it's put in `dev-dependencies`: 

```rust
#[cfg_attr(test, mockall_double::double)]
crate::MyThing;
```

Can anyone clarify if there is any potential issue if doing it this way and keep mockall_double in dev-dep?

**Comment by asomers** [comment, marked as answer]
There's little point to doing that.  The whole reason that mockall_double is its own crate, rather than part of mockall, is so it can be made a full dependency without pulling in too many transitive dependencies.  But if you REALLY don't want your crate to depend on syn or quote in release builds, then what you propose is totally ok, too.

**Reply by oaijiao** [reply]
Got it. Thanks for the quick response and clarification!
